matcher.go

```diff
@@ -47,6 +93,48 @@ func MatchRegexp(r *regexp.Regexp) MatcherFunc {
 	}
 }
 
+// MatchTimeStamp returns a MatcherFunc that matches a time layout pattern in given string
+func MatchTimeStamp(layout string) MatcherFunc{
```

Comment:
Since [timestamp](https://dictionary.cambridge.org/us/dictionary/english/timestamp) is single work, function name `MatchTimestamp` will be more appropriate.